### PR TITLE
Freezer

### PR DESCRIPTION
**Pull request by Amelia0010**
Fairly certain this used to be a thing back in the day. Either way:

Freezer now protects you from both regular explosions, and also the nuke exploding since it uses a snowflake proc rather than just using regular explosions

:cl:
tweak: you'll now survive the nuclear bomb exploding if you're in a freezer
/:cl:

### CODE REVIEW

**Comment by AshCorr** [comment]
powercreep

**Comment by jcatjmeow** [comment]
this was a prebase feature

indiana jone

**Comment by ToGWtF** [comment]
This still works, I use it on nuke ops regularly. The code is handled instead in `Yogstation/code/game/objects/structures/crates_lockers/closets/secure/freezer.dm`

**Comment by Amelia0010** [comment]
that's onyl explosions not nukes :)

File: code/modules/antagonists/nukeop/equipment/nuclearbomb.dm [hunk 1]
```diff
@@ -578,11 +578,18 @@
...
+  *
+  * Arguments:
+  * * z - the z-level to kill people on
+  */
 /proc/KillEveryoneOnZLevel(z)
 	if(!z)
 		return
 	for(var/mob/M in GLOB.mob_list)
-		if(M.stat != DEAD && M.z == z)
+		if(M.stat != DEAD && M.z == z && !istype(M.loc, /obj/structure/closet/secure_closet/freezer))
```
Comment (alexkar598) [inline code/modules/antagonists/nukeop/equipment/nuclearbomb.dm]: You realize that if you are in a freezer, you are in nullspace, so you dont exist on the Z level making your check redundant as the M.z == z check will prevent anyone in a locker or other container from dying, perhaps you might wanna fix that?
```suggestion
		if(M.stat != DEAD && get_turf(M).z == z && !istype(M.loc, /obj/structure/closet/secure_closet/freezer))
```

**Review by alexkar598 — DISMISSED** [review]

**Review by fluffe9911 — APPROVED** [review]
nice

**Pull request merged**